Changes for the mapper for StepTemplateDTO

WesleyAnderson349 opened the pull request.

WesleyAnderson349 committed:
Changes for the mapper for StepTemplateDTO

timothy-stratton-cairn commented on src/main/java/com/cairn/waypoint/dashboard/endpoints/steptemplate/dto/StepTemplateDetailsDto.java:
```diff
@@ -12,7 +12,7 @@ public class StepTemplateDetailsDto {
   private String description;
   private String status;
-  private StepTaskDetailsDto linkedStepTask;
+  private StepTaskDetailsDto linkedStepTask; // mapping to this with reflection. 
```
We can remove this comment here

timothy-stratton-cairn commented on src/main/java/com/cairn/waypoint/dashboard/WaypointDashboardApiApp.java:
```diff
@@ -22,7 +22,7 @@ public WaypointDashboardApiApp(List<StatusObserver> protocolStatusObservers) {
   public static void main(String[] args) {
     SpringApplication app = new SpringApplication(WaypointDashboardApiApp.class);
-    app.setAdditionalProfiles("default");
+    app.setAdditionalProfiles("dev");
```
Let's revert this back to to `default`. This file should more often than not remain unchanged. When you do your commits in the future, likely you can exclude this from staging.

timothy-stratton-cairn commented on .gitignore:
```diff
@@ -25,3 +25,9 @@ replay_pid*
+.project
+.settings/org.eclipse.core.resources.prefs
+.settings/org.eclipse.jdt.apt.core.prefs
+.settings/org.eclipse.jdt.core.prefs
```
Let's add the entire `bin/*` directory into this .gitignore so that we can remove these files out of the repo

timothy-stratton-cairn commented on src/main/java/com/cairn/waypoint/dashboard/endpoints/steptemplate/dto/HomeworkTemplateDetailsDto.java:
```diff
@@ -9,4 +14,13 @@ public class HomeworkTemplateDetailsDto {
+}
+
+@Mapper
+interface StepTemplateLinkedHomeworkTemplateMapper {
```
I see what you've done here. You've utilized a mapper here inside the class. This could be a maintenance hotspot, especially on larger teams/as the team grows. Could we instead move the code here into a separate `*.java` file in the adjacent `.mapper` package?

timothy-stratton-cairn commented on src/main/resources/application-dev.yml:
```diff
@@ -1,6 +1,6 @@
 waypoint:
   dashboard:
-    base-url: http://host.docker.internal:8083
```
Let's revert the content of this file also. You can ignore pushing up this file in the future as it contains the configs specific to your system

WesleyAnderson349 committed:
Just so I don't lose my work

WesleyAnderson349 committed:
safety commit

WesleyAnderson349 committed:
quick update

WesleyAnderson349 committed:
More changes

deleted a bunch of the files that were commented out, changes to GetALLQuestionsAndResponseByUser Endpoint
Created a GetAllProtocolsByUserID endpoint
Required DataService and Repository Changes

WesleyAnderson349 committed:
Simplified dtos

Simplified Protocol,Question, Response DTOs so we're not pushing everything on the GetQuestionsResponse end point

WesleyAnderson349 commented:
This was supposed to go up yesterday 
Back end changes

WesleyAnderson349 committed:
Bunch of changes:

Got rid of Protocol on Responses and anything that needed it.
AddHomeworkResponseEndpoint
LinkHomeworkQuestionAndProtocolTemplate Endpoint
GetHomeworkByProtocol/ProtocolTemplate Endpoints Added
+ Required Changes to DTOs and DataService/Repositories.

timothy-stratton-cairn commented on src/main/java/com/cairn/waypoint/dashboard/endpoints/homework/UpdateHomeworkResponsesEndpoint.java:
```diff
@@ -47,144 +43,110 @@
 
-  public static final String PATH = "/api/homework/{homeworkId}";
-  private final HomeworkDataService homeworkDataService;
+  public static final String PATH = "/api/protocol/{protocolId}";
```
This path should be `/api/homework-response/{homeworkResponseId}`

timothy-stratton-cairn commented on src/main/java/com/cairn/waypoint/dashboard/endpoints/homeworkquestion/DeleteHomeworkQuestionByIdEndpoint.java:
```diff
@@ -32,14 +33,17 @@ public class DeleteHomeworkQuestionByIdEndpoint {
   public static final String PATH = "/api/homework-question/{homeworkQuestionId}";
   private final HomeworkQuestionDataService homeworkQuestionDataService;
-  private final HomeworkTemplateLinkedHomeworkQuestionDataService homeworkTemplateLinkedHomeworkQuestionDataService;
+  //private final HomeworkTemplateLinkedHomeworkQuestionDataService homeworkTemplateLinkedHomeworkQuestionDataService;
```
Let's remove this comment

timothy-stratton-cairn commented on src/main/java/com/cairn/waypoint/dashboard/endpoints/homework/DeleteHomeworkByIdEndpoint.java:
```diff
@@ -39,7 +39,7 @@ public DeleteHomeworkByIdEndpoint(HomeworkDataService homeworkDataService,
     this.protocolStepLinkedHomeworkService = protocolStepLinkedHomeworkService;
   }
-
+/*
```
Let's remove all commented code

WesleyAnderson349 committed:
Small changes to set up for dev server update

timothy-stratton-cairn commented on .externalToolBuilders/org.springframework.ide.eclipse.boot.validation.springbootbuilder.launch:
Let's add this file/hidden folder to `.gitignore`

timothy-stratton-cairn commented on .settings/org.eclipse.m2e.core.prefs:
you can probably just add `.settings/` to your git ignore to cover all the files under this folder as opposed to adding them individually

timothy-stratton-cairn commented on pom.xml:
```diff
@@ -41,7 +41,13 @@
         <version>3.12.1</version>
       </plugin>
-    </plugins>
+	<plugin>
```
I think we can remove this. We resolved your liquibase issues by bringing up the project out of the eclipse workspace into separate intellij projects

timothy-stratton-cairn commented on pom.xml:
```diff
@@ -160,6 +166,11 @@
       <groupId>org.apache.poi</groupId>
       <version>${poi.version}</version>
     </dependency>
+    <dependency>
```
let's definitely remove this. This only resolves because authorization-api lives locally in your .m2 directory. It won't be able to download this from maven central at which all dependencies live

timothy-stratton-cairn commented on src/main/java/com/cairn/waypoint/dashboard/endpoints/HomeworkQuestionLinkedProtocolTemplate/AddHomeworkQuestionLinkedProtocolTemplateDto.java:
```diff
@@ -0,0 +1,22 @@
+    private Long protocolTemplateId;
+    private Long homeworkQuestionId;
+
+    public Long getProtocolTemplateId() {
```
Let's remove these explicit getters and setters to utilize the paradigm I've setup in our other DTOs. Check them out and you will see a series of Lombok annotations that generate getters, setters, equals, hashCode, and other methods for our POJO classes

timothy-stratton-cairn commented on src/main/java/com/cairn/waypoint/dashboard/endpoints/HomeworkQuestionLinkedProtocolTemplate/HomeworkQuestionLinkedProtocolTemplateDto.java:
```diff
@@ -0,0 +1,12 @@
+import lombok.Builder;
+import lombok.Data;
+
+@Data
```
Kind of like you've done here with @Data. This will generate your getters/setters/etc

timothy-stratton-cairn commented on src/main/java/com/cairn/waypoint/dashboard/endpoints/HomeworkQuestionLinkedProtocolTemplate/LinkHomeworkQuestionAndProtocolTemplateEndpoint.java:
```diff
@@ -0,0 +1,94 @@
+
+@Slf4j
+@RestController
+@RequestMapping("/api/protocol-template-homework")
```
This paradigm is not how we setup the request mapping. Could you add this to the `@PostMapping` on the method below instead of at the class level?

timothy-stratton-cairn replied:
Also, this request path should look more like this: `POST api/protocol-template/homework-question`. We are adding a homework question to a protocol template

timothy-stratton-cairn commented on src/main/java/com/cairn/waypoint/dashboard/endpoints/HomeworkQuestionLinkedProtocolTemplate/LinkHomeworkQuestionAndProtocolTemplateEndpoint.java:
```diff
@@ -0,0 +1,94 @@
+                .question(homeworkQuestionOpt.get())
+                .build();
+
+        HomeworkQuestionLinkedProtocolTemplate savedEntry = linkedProtocolTemplateRepository.save(linkedEntry);
```
We should not be interacting with the Repository layer in the Controller. The entry to be saved should be passed via the Service Layer. Please add a service for the protocolTemplate (or update the existing) and create a method that does this. This keeps a layer of abstraction between the data layer and the controller layer

timothy-stratton-cairn commented on src/main/java/com/cairn/waypoint/dashboard/endpoints/homework/UpdateHomeworkResponsesEndpoint.java:
```diff
@@ -47,144 +43,110 @@
-      S3FileUpload s3FileUpload, ProtocolStepDataService protocolStepDataService,
-      ProtocolDataService protocolDataService) {
-    this.homeworkDataService = homeworkDataService;
+  public UpdateHomeworkResponsesEndpoint(S3FileUpload s3FileUpload, ProtocolDataService protocolDataService,HomeworkResponseDataService responseDataService ) {
```
Make sure your add spaces between arguments for readability. I should send you the style guide I use so that it does this for you on save

timothy-stratton-cairn commented on src/main/java/com/cairn/waypoint/dashboard/endpoints/homework/dto/UpdateHomeworkResponseDetailsListDto.java:
```diff
@@ -1,6 +1,9 @@
 
 import java.util.List;
+
+//import com.cairn.waypoint.dashboard.entity.Homework;
```
Can remove this commented code

timothy-stratton-cairn commented on src/main/java/com/cairn/waypoint/dashboard/endpoints/homeworkquestion/GetHomeworkQuestionByCategoryEndpoint.java:
```diff
@@ -0,0 +1,92 @@
+	  }
+
+
+@GetMapping(PATH)
```
Check your spacing/indenting here. This is off. We can prioritize the style guide next meeting

timothy-stratton-cairn commented on src/main/java/com/cairn/waypoint/dashboard/endpoints/homeworkquestion/GetHomeworkQuestionByCategoryEndpoint.java:
```diff
@@ -0,0 +1,92 @@
+            content = {@Content(schema = @Schema(hidden = true))}),
+        @ApiResponse(responseCode = "403", description = "Forbidden",
+            content = {@Content(schema = @Schema(hidden = true))})})
+
```
Shouldn't be any spaces between our annotations and their owning methods/properties

timothy-stratton-cairn commented on src/main/java/com/cairn/waypoint/dashboard/endpoints/homeworkquestion/GetHomeworkQuestionByProtocolIdEndpoint.java:
```diff
@@ -0,0 +1,92 @@
+                            content = {@Content(schema = @Schema(hidden = true))}),
+                    @ApiResponse(responseCode = "403", description = "Forbidden",
+                            content = {@Content(schema = @Schema(hidden = true))})})
+
```
Same here, remove the whitespace

timothy-stratton-cairn commented on src/main/java/com/cairn/waypoint/dashboard/endpoints/homeworkquestion/GetHomeworkQuestionByTemplateIdEndpoint.java:
```diff
@@ -0,0 +1,92 @@
+
+
+
+@GetMapping(PATH)
```
Spacing/indenting here

timothy-stratton-cairn commented on src/main/java/com/cairn/waypoint/dashboard/endpoints/homeworkquestion/GetHomeworkQuestionByTemplateIdEndpoint.java:
```diff
@@ -0,0 +1,92 @@
+
+@Slf4j
+@RestController
+@Tag(name = "Homework Question d")
```
Typo here -- there is a trailing `d`

timothy-stratton-cairn commented on src/main/java/com/cairn/waypoint/dashboard/endpoints/homeworkquestion/GetHomeworkQuestionByCategoryEndpoint.java:
```diff
@@ -0,0 +1,92 @@
+
+@Slf4j
+@RestController
+@Tag(name = "Homework")
```
I think this tag should be `Homework Question` instead

timothy-stratton-cairn commented on src/main/java/com/cairn/waypoint/dashboard/endpoints/homeworkquestion/GetHomeworkQuestionResponseByCategoryEndpoint.java:
```diff
@@ -0,0 +1,89 @@
+
+@Slf4j
+@RestController
+@Tag(name = "Homework")
```
I think this tag should be `Homework Question` instead

timothy-stratton-cairn commented on src/main/java/com/cairn/waypoint/dashboard/endpoints/homeworkquestion/GetHomeworkQuestionResponseByCategoryEndpoint.java:
```diff
@@ -0,0 +1,89 @@
+	  }
+
+
+@GetMapping(PATH)
```
Spacing here

timothy-stratton-cairn commented on src/main/java/com/cairn/waypoint/dashboard/endpoints/homeworkquestion/GetHomeworkQuestionResponseByCategoryEndpoint.java:
```diff
@@ -0,0 +1,89 @@
+            content = {@Content(schema = @Schema(hidden = true))})})
+
+
+public ResponseEntity<?> getAllResponseBycategoryId(@PathVariable Long categoryId,
```
Spacing and indentation here. Whitespace also

timothy-stratton-cairn commented on src/main/java/com/cairn/waypoint/dashboard/endpoints/homeworkquestion/GetHomeworkQuestionsAndResponsesByUserEndpoint.java:
```diff
@@ -0,0 +1,86 @@
+
+@Slf4j
+@RestController
+@Tag(name = "Homework Question")
```
Check your domain here -- should this be `Homework Question Response`?

timothy-stratton-cairn commented on src/main/java/com/cairn/waypoint/dashboard/endpoints/homeworkquestion/GetHomeworkResponseByProtocolIdEndpoint.java:
```diff
@@ -0,0 +1,87 @@
+
+@Slf4j
+@RestController
+@Tag(name = "Homework")
```
`Homework Question Response` to be consistent with your other endpoint

timothy-stratton-cairn commented on src/main/java/com/cairn/waypoint/dashboard/endpoints/homeworkquestion/GetHomeworkResponseByProtocolIdEndpoint.java:
```diff
@@ -0,0 +1,87 @@
+	  }
+
+
+@GetMapping(PATH)
```
Won't comment any further on spacing, indentation, and whitespace but note these going forward and resolve this generally

timothy-stratton-cairn commented on src/main/java/com/cairn/waypoint/dashboard/endpoints/homeworkquestion/JoinQuestionToProtocolTemplate.java:
```diff
@@ -0,0 +1,85 @@
+
+@Slf4j
+@RestController
+@Tag(name = "Homework")
```
`Protocol Template`

timothy-stratton-cairn commented on src/main/java/com/cairn/waypoint/dashboard/endpoints/homeworkquestion/SimplifiedHomeworkQuestionDto.java:
```diff
@@ -0,0 +1,61 @@
+
+import java.time.LocalDateTime;
+
+public class SimplifiedHomeworkQuestionDto {
```
Let's use the annotated approach as opposed to explicit getters/setters here. You can utilize `@Data` to generate these via Lombok

timothy-stratton-cairn replied:
It will also create your constructor for you. Let's delete all constructors, getters, and setters from all POJOs going forward

timothy-stratton-cairn replied:
You could use a `Mapper` with Mapstruct to map a homeworkQuestion to the simplifiedHomeworkQuestion

timothy-stratton-cairn commented on src/main/java/com/cairn/waypoint/dashboard/endpoints/homeworkquestion/dto/AddLinkedQuestionProtocolTemplateDto.java:
```diff
@@ -0,0 +1,19 @@
+import lombok.Data;
+
+
+@Data
```
good use of `@Data` here. But check your spacing on your annotations below

timothy-stratton-cairn commented on src/main/java/com/cairn/waypoint/dashboard/endpoints/homeworkresponse/dto/AddHomeworkQuestionResponseDto.java:
```diff
@@ -0,0 +1,55 @@
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import org.springframework.web.multipart.MultipartFile;
+
+public class AddHomeworkQuestionResponseDto {
```
Getters and setters once again.

timothy-stratton-cairn commented on src/main/java/com/cairn/waypoint/dashboard/endpoints/homeworkresponse/dto/AddHomeworkResponseEndpoint.java:
```diff
@@ -0,0 +1,133 @@
+
+    private final HomeworkResponseDataService homeworkResponseDataService;
+    private final HomeworkQuestionDataService homeworkQuestionDataService;
+    private final HomeworkCategoryRepository homeworkCategoryRepository;
```
Shouldn't be using repositories in your controller. Should be abstracted via the Service layer

timothy-stratton-cairn commented on src/main/java/com/cairn/waypoint/dashboard/endpoints/homeworkresponse/dto/AddHomeworkResponseEndpoint.java:
```diff
@@ -0,0 +1,133 @@
+                            addHomeworkResponseDto.getQuestionId() + "] does not exist",
+                    HttpStatus.NOT_FOUND);
+        }
+        Optional<HomeworkCategory> categoryOpt = homeworkCategoryRepository.findById(addHomeworkResponseDto.getCategoryId());
```
Refactor to use a service call instead

timothy-stratton-cairn commented on src/main/java/com/cairn/waypoint/dashboard/endpoints/homeworkresponse/dto/AddHomeworkResponseEndpoint.java:
```diff
@@ -0,0 +1,133 @@
+
+        HomeworkResponse createdResponse = homeworkResponseDataService.saveHomeworkResponse(homeworkResponse);
+
+        HomeworkResponseDto responseDto = HomeworkResponseDto.builder()
```
So note that each of these could be a single mapstruct call as opposed to the builder paradigm. This could be a  refactor for later but note there is a more concise way to construct these DTOs

timothy-stratton-cairn commented on src/main/java/com/cairn/waypoint/dashboard/endpoints/homeworkresponse/dto/HomeworkResponseDto.java:
```diff
@@ -0,0 +1,48 @@
+
+@Data
+@Builder
+public class HomeworkResponseDto{
```
You annotated this class but still explicitly implemented your getters and setters here. Let's remove those and let the annotation do the work

timothy-stratton-cairn commented on src/main/java/com/cairn/waypoint/dashboard/endpoints/homeworkresponse/dto/HomeworkResponseListDto.java:
```diff
@@ -0,0 +1,19 @@
+	List<HomeworkResponse> responses;
+	private Integer numberOfResponses;
+	
+    public Integer getNumOfQuestions() {
```
Should this be `getNumOfResponses()` instead?

timothy-stratton-cairn commented on src/main/java/com/cairn/waypoint/dashboard/endpoints/homeworkresponse/dto/SimplifiedHomeworkResponseDto.java:
```diff
@@ -0,0 +1,42 @@
+public class SimplifiedHomeworkResponseDto {
+    private Long id;
+
+    public String getResponse() {
```
The structure of this class is a bit off. You have properties interspersed throughout with methods implemented within. Could we refactor to use lombok instead and keep our properties at the top of the class.

timothy-stratton-cairn commented on src/main/java/com/cairn/waypoint/dashboard/endpoints/homeworktemplate/DeleteHomeworkTemplateByIdEndpoint.java:
```diff
@@ -1,12 +1,8 @@
-import com.cairn.waypoint.dashboard.service.data.HomeworkTemplateDataService;
-import com.cairn.waypoint.dashboard.service.data.ProtocolStepLinkedHomeworkService;
-import com.cairn.waypoint.dashboard.service.data.StepTemplateLinkedHomeworkTemplateDataService;
+//import com.cairn.waypoint.dashboard.service.data.HomeworkTemplateDataService;
```
let's remove commented code here

timothy-stratton-cairn commented on src/main/java/com/cairn/waypoint/dashboard/endpoints/protocol/AddProtocolEndpoint.java:
```diff
@@ -64,14 +62,12 @@ public class AddProtocolEndpoint {
       StepTemplateDataService stepTemplateDataService,
-      HomeworkDataService homeworkDataService, HouseholdDataService householdDataService,
-      ProtocolStepLinkedHomeworkService protocolStepLinkedHomeworkService) {
+     //HomeworkDataService homeworkDataService, 
```
Let's remove commented code here

timothy-stratton-cairn commented on src/main/java/com/cairn/waypoint/dashboard/endpoints/protocol/AddProtocolEndpoint.java:
```diff
@@ -188,8 +184,8 @@ public ResponseEntity<?> assignProtocol(
 
-      protocolTemplateHelperService.generateAndSaveClientHomework(createdProtocol,
-          principal.getName());
+      //protocolTemplateHelperService.generateAndSaveClientHomework(createdProtocol,
```
And here

timothy-stratton-cairn commented on src/main/java/com/cairn/waypoint/dashboard/endpoints/protocol/DeleteProtocolEndpoint.java:
```diff
@@ -81,10 +81,10 @@ public ResponseEntity<String> generateSuccessResponse(
         note.setModifiedBy(modifiedBy);
       });
-      step.getLinkedHomework().forEach(homeworkLink -> {
+        /*step.getLinkedHomework().forEach(homeworkLink -> {
```
Will forego adding the commented code removal comment but let's do this globally. You could do a global search in the project for `//` or `/*` to find and remove all your comments within the codebase.

timothy-stratton-cairn commented on src/main/java/com/cairn/waypoint/dashboard/endpoints/protocol/SimplifiedProtocolDto.java:
```diff
@@ -0,0 +1,49 @@
+
+import com.cairn.waypoint.dashboard.entity.Protocol;
+
+public class SimplifiedProtocolDto {
```
Lombok use here. Removing the DTOs also

timothy-stratton-cairn commented on src/main/java/com/cairn/waypoint/dashboard/endpoints/protocoltemplate/AppendStepTemplateToProtocolTemplateEndpoint.java:
```diff
@@ -43,14 +43,18 @@ public class AppendStepTemplateToProtocolTemplateEndpoint {
-      ProtocolDataService protocolDataService, HomeworkDataService homeworkDataService,
-      ProtocolStepLinkedHomeworkService protocolStepLinkedHomeworkService) {
+      ProtocolDataService protocolDataService
+      //HomeworkDataService homeworkDataService,
```
commented code

timothy-stratton-cairn commented on src/main/java/com/cairn/waypoint/dashboard/entity/HomeworkQuestionLinkedProtocolTemplate.java:
```diff
@@ -0,0 +1,40 @@
+	
+	@Id
+	@GeneratedValue(strategy = GenerationType.IDENTITY)
+	private Long id;
```
Does this table have an `id` field? If so, let's remove it. There is no need for it and this becomes bloatware. We can then remove this field. Note that in your liquibase you should ADD a new yaml file to remove the column and then remove the field from this class. That should resolve this.

timothy-stratton-cairn commented on src/main/java/com/cairn/waypoint/dashboard/entity/HomeworkResponseLinkedProtocol.java:
```diff
@@ -16,21 +19,22 @@
+	
+	@Id
+	@GeneratedValue(strategy = GenerationType.IDENTITY)
+	private Long id;
```
Same, let's remove this as it will result in bloatware. There isn't a usage that I can foresee and we will need to maintain it longterm if it is left in now

timothy-stratton-cairn commented on src/main/java/com/cairn/waypoint/dashboard/entity/StepTemplateLinkedHomeworkTemplate.java:
```diff
@@ -13,22 +13,23 @@
+//@AllArgsConstructor
+//@SQLRestriction("active=1")
+//@Table(name = "step_template_linked_homework_template")
 public class StepTemplateLinkedHomeworkTemplate extends BaseEntity {
```
Can probably delete this class now that the table has been deleted.

timothy-stratton-cairn commented on src/main/java/com/cairn/waypoint/dashboard/repository/HomeworkQuestionRepository.java:
```diff
@@ -9,7 +9,12 @@
   Optional<HomeworkQuestion> findByQuestion(String question);
-
+  
+  
```
Be sure to delete unnecessary whitespace

timothy-stratton-cairn commented on src/main/java/com/cairn/waypoint/dashboard/service/data/HomeworkQuestionDataService.java:
```diff
@@ -1,19 +1,31 @@
   private final HomeworkQuestionRepository homeworkQuestionRepository;
+  private final HomeworkQuestionLinkedProtocolTemplatesRepository homeworkQuestionLinkedProtocolTemplatesRepository;
+  private final ProtocolTemplateRepository protocolTemplateRepository;
+  private final ProtocolDataService protocolDataService;
```
One must be careful not to overuse other services within a service. If there is common functionality, lean of the rule of 3 and copy said functionality until there are three instance. Then you build a `Base**Service` to share that functionality. Using one service method within another service could result in unforeseen changes when one domain changes. You're introducing a coupling between the domains/verticals that could make maintainability a problem going forward. Instead here, you could just use the repository call.

timothy-stratton-cairn commented on src/main/java/com/cairn/waypoint/dashboard/service/data/HomeworkQuestionLinkedProtocolTemplateDataService.java:
```diff
@@ -0,0 +1,58 @@
+
+	        return linkedRepository.save(link);
+	    }
+	 public List<HomeworkQuestion> findAllQuestionsByQuestionId(Long questionId) {
```
Should be whitespace between method bodies and subsequent signatures

timothy-stratton-cairn commented on src/main/java/com/cairn/waypoint/dashboard/service/data/HomeworkResponseDataService.java:
```diff
@@ -1,25 +1,58 @@
+          ProtocolRepository protocolRepository,
+          HomeworkQuestionRepository questionRepository,
+          HomeworkQuestionLinkedProtocolTemplatesRepository linkedRepository,
+          ProtocolDataService protocolDataService,
```
Again calls to sibling services should be avoided within services.

timothy-stratton-cairn commented on src/main/java/com/cairn/waypoint/dashboard/service/data/QuestionResponsePairDataService.java:
```diff
@@ -0,0 +1,37 @@
+
+    public QuestionResponsePairDataService( HomeworkQuestionDataService questionService,HomeworkResponseDataService homeworkResponseDataService){
+        this.questionService = questionService;
+        this.homeworkResponseDataService = homeworkResponseDataService;
```
Services calls from one service. I might be lenient in this particular service as it is not a domain service like `ProtocolDataService` or `HomeworkDataService` but note that this is a unique situation here.

timothy-stratton-cairn commented on src/main/java/com/cairn/waypoint/dashboard/service/helper/HomeworkQuestionHelperService.java:
```diff
@@ -14,6 +21,32 @@ public HomeworkQuestionHelperService(ExpectedResponseDataService expectedRespons
   }
 
+  public HomeworkQuestionDto generateHomeworkQuestionDto(HomeworkQuestion question) {
+	  return HomeworkQuestionDto.builder()
```
Again note your use of the builder paradigm when a mapper could do this for you.

timothy-stratton-cairn commented on src/main/java/com/cairn/waypoint/dashboard/service/helper/HomeworkQuestionResponseHelperService.java:
```diff
@@ -0,0 +1,24 @@
+  }
+  
+  public UpdateHomeworkResponseDetailsDto generateHomeworkQuestionResponseDto(HomeworkResponse response) {
+	  return UpdateHomeworkResponseDetailsDto.builder()
```
Mapper instead of builder.

timothy-stratton-cairn commented on src/main/java/com/cairn/waypoint/dashboard/service/helper/HomeworkQuestionResponseHelperService.java:
```diff
@@ -0,0 +1,24 @@
+import com.cairn.waypoint.dashboard.service.data.ExpectedResponseDataService;
+
+@Service
+@SuppressWarnings("DuplicatedCode")
```
If there is duplicated code, try to refactor so that you don't need to suppress it. This could be discretionary but we can discuss

timothy-stratton-cairn commented on src/main/resources/db/changelog/2024-10-15-add-label-to-homework-question-table.yaml:
```diff
@@ -0,0 +1,11 @@
+      changes:
+        - sql:
+            dbms: 'h2, mysql'
+            sql: alter table homework_question add `label` varchar(36) default NULL after `question`;
```
Probably too late for your local changelog checksums but note your syntax here. keywords should be CAPITALIZED

WesleyAnderson349 committed:
Simple branch changes No refactor

timothy-stratton-cairn commented on .gitignore:
```diff
@@ -28,6 +28,11 @@ target/
+.externalToolBuilders/org.springframework.ide.eclipse.boot.validation.springbootbuilder.launch
+
+#
+src/main/resources/application-dev.yml
```
So this shouldn't be ignored particularly. It should just be uncommitted locally. Adding it here removes it from source control and we do indeed need it inside the repository

WesleyAnderson349 committed:
Update .gitignore

WesleyAnderson349 committed:
Data Service changes

WesleyAnderson349 committed:
AddHomeworkRefactor

to avoid Repositories in endpoints
+ Category Dataservice
+ QuestionPairData Service to not use other repositories

WesleyAnderson349 committed:
Simplified DTOs

Changes to use Lombok more

WesleyAnderson349 committed:
Bunch of line deletes

Mostly getting rid of bad import statements but I'm pretty sure I deleted these a while back.
Might have been caused when I tried to switch branches.

timothy-stratton-cairn committed:
Update .gitignore

timothy-stratton-cairn committed:
Delete .externalToolBuilders/org.springframework.ide.eclipse.boot.validation.springbootbuilder.launch

timothy-stratton-cairn committed:
Delete .settings/org.springframework.ide.eclipse.prefs

timothy-stratton-cairn committed:
Delete .settings/org.eclipse.m2e.core.prefs

WesleyAnderson349 committed:
pom.xml

WesleyAnderson349 committed:
pomxml and ApiApp changes

WesleyAnderson349 committed:
Merge branch 'wesTest' of https://github.com/timothy-stratton-cairn/waypoint-dashboard-api into wesTest

WesleyAnderson349 committed:
Spacing Fix

WesleyAnderson349 committed:
Spacing/Style Fixes

WesleyAnderson349 committed:
Delete comments changes

WesleyAnderson349 committed:
Simple mappers+ Category stuff and Fix to GetHomeworkQuestionByProtocolIdEndpoint
